pkg/storkctl/clusterpair.go

```diff
@@ -273,3 +275,193 @@ func newGenerateClusterPairCommand(cmdFactory Factory, ioStreams genericclioptio
+			if err != nil {
+				util.CheckErr(err)
+				return
+			}
+
+			// Create cluster-pair on source cluster
+			conf, err := getConfig(sFile).ClientConfig()
+			if err != nil {
+				util.CheckErr(err)
+				return
+			}
+			storkops.Instance().SetConfig(conf)
+			_, err = storkops.Instance().CreateClusterPair(srcClusterPair)
+			if err != nil {
+				util.CheckErr(err)
+				return
+			}
+			printMsg("ClusterPair "+clusterPairName+" created successfully on source cluster", ioStreams.Out)
+			// Create cluster-pair on dest cluster
+			conf, err = getConfig(dFile).ClientConfig()
+			if err != nil {
+				util.CheckErr(err)
+				return
+			}
+			storkops.Instance().SetConfig(conf)
+			_, err = storkops.Instance().CreateClusterPair(destClusterPair)
+			if err != nil {
+				util.CheckErr(err)
+				return
+			}
+			printMsg("ClusterPair "+clusterPairName+" created successfully on destination cluster", ioStreams.Out)
+
+		},
+	}
+
+	createClusterPairCommand.Flags().StringVarP(&sIP, "srcip", "", "", "ip of storage node from source cluster")
+	createClusterPairCommand.Flags().StringVarP(&sPort, "srcport", "", "9001", "port of storage node from source cluster")
+	createClusterPairCommand.Flags().StringVarP(&sFile, "src-kube-file", "", "", "kube-config of source cluster")
+	createClusterPairCommand.Flags().StringVarP(&dIP, "destip", "", "", "kube-config of destination cluster")
+	createClusterPairCommand.Flags().StringVarP(&dPort, "destport", "", "9001", "port of storage node from destination cluster")
+	createClusterPairCommand.Flags().StringVarP(&dFile, "dest-kube-file", "", "", "kube-config of destination cluster")
+	createClusterPairCommand.Flags().StringVarP(&srcToken, "srctoken", "", "", "source cluster token for cluster pairing")
+	createClusterPairCommand.Flags().StringVarP(&destToken, "desttoken", "", "", "destination cluster token for cluster pairing")
+
+	return createClusterPairCommand
+}
+
+func generateClusterPair(name, ns, ip, port, token, configFile string) (*storkv1.ClusterPair, error) {
+	opts := make(map[string]string)
+	opts["ip"] = ip
+	opts["port"] = port
+	// extract token from px-endpoint command
+	opts["token"] = token
+
+	config, err := getConfig(configFile).RawConfig()
+	if err != nil {
+		return nil, err
+	}
+	currConfig, err := pruneConfigContexts(config)
+	if err != nil {
+		return nil, err
+	}
+	clusterPair := &storkv1.ClusterPair{
+		TypeMeta: meta.TypeMeta{
+			Kind:       reflect.TypeOf(storkv1.ClusterPair{}).Name(),
+			APIVersion: storkv1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: meta.ObjectMeta{
+			Name:      name,
+			Namespace: ns,
+		},
+
+		Spec: storkv1.ClusterPairSpec{
+			Config:  currConfig,
+			Options: opts,
+		},
+	}
+	return clusterPair, nil
+}
+
+// Prune out all but the current-context and related
+// info
+func pruneConfigContexts(config clientcmdapi.Config) (clientcmdapi.Config, error) {
```

Comment:
Same code is being used in `newGenerateClusterPairCommand`, this can be called from there too